beacon/goclient/events.go

```diff
@@ -0,0 +1,88 @@
+package goclient
+
+import (
+	"context"
+
+	apiv1 "github.com/attestantio/go-eth2-client/api/v1"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type event interface {
+	*apiv1.HeadEvent
+}
+
+type subscriber[T event] struct {
+	Identifier string
+	Channel    chan T
+}
+
+func (gc *GoClient) SubscribeToHeadEvents(ctx context.Context, subscriberIdentifier string) (<-chan *apiv1.HeadEvent, error) {
```

Comment:
it's better to let callers provide the channel so they can control their buffer size